File: 프로그래머스/2/42747. H－Index/H－Index.cs
using System;

public class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        int n = citations.Length;
        Array.Sort(citations);
        
        for(int i = 0; i < n; i++)
        {
            int hIndex = n - i;
            if(hIndex <= citations[i])
            {
                answer = hIndex;
                break;
            }
        }
        return answer;
    }
}
// 0 1 3 5 6
// 0 5
// 1 4
// 2 3
// 3 3
// 4 2

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;
// 6 5 3 1 0
int solution(vector<int> citations) {
    int answer = 0;
    sort(citations.begin(), citations.end(), greater<int>());
    
    for(int i = 0; i < citations.size(); i++)
    {
        if(citations[i] >= i + 1)
            answer = i + 1;
        else
            break;
    }
    return answer;
}

// 